using-rest-api.md updated

jesusgonzalez12 opened the pull request:
This version introduces improvements on how a user can work with edge collections and graphs using the pyc8 SDK.
Only changes to the pyc8 code have been introduced.
For more details, you can check the Kanban board.

jesusgonzalez12 committed:
using-rest-api.md updated

This version introduces improvements on how a user can work with edge collections and graphs using the pyc8 SDK.
Only changes to the pyc8 code have been introduced.
For more details, you can check the Kanban board.

jamesm-macrometa commented on docs/collections/graphs/using-rest-api.md:
```diff
@@ -60,193 +60,196 @@ To create `edge collection` use same endpoint `/_fabric/{fabric_name}/_api/colle
+FEDERATION = "api-gdn.paas.macrometa.io"
+FED_URL = f"https://{FEDERATION}"
+GEO_FABRIC = "_system"
+API_KEY = "my IP key" #Change to my API key
```
```suggestion
API_KEY = "my IP key" # Change to your API key
```

jamesm-macrometa commented on docs/collections/graphs/using-rest-api.md:
```diff
@@ -60,193 +60,196 @@ To create `edge collection` use same endpoint `/_fabric/{fabric_name}/_api/colle
+API_KEY = "my IP key" #Change to my API key
+#EMAIL = "my email"
+#PASSWORD = "my password"
+#AUTH_TOKEN = "bearer " #do not change
```
```suggestion
#AUTH_TOKEN = "bearer " # Do not change
```

jamesm-macrometa commented on docs/collections/graphs/using-rest-api.md:
```diff
@@ -60,193 +60,196 @@ To create `edge collection` use same endpoint `/_fabric/{fabric_name}/_api/colle
+GRAPH_NAME = "lectureteacher"
+
+# Create a HTTPS Session Part 1
+# This part is needed when getting access through email/pass instead of API key or JWT token
```
```suggestion
# The following is required when accessing Macrometa with your email and password instead of an API key or JWT token
```

jamesm-macrometa commented on docs/collections/graphs/using-rest-api.md:
```diff
@@ -60,193 +60,196 @@ To create `edge collection` use same endpoint `/_fabric/{fabric_name}/_api/colle
+session.headers.update({"content-type": 'application/json'})
+session.headers.update({"authorization": "apikey " + API_KEY})
+
+# Create Doc Collections and Insert Data to Document Collections
```
```suggestion
# Create Document Collections and Insert Data
```

jamesm-macrometa commented on docs/collections/graphs/using-rest-api.md:
```diff
@@ -60,193 +60,196 @@ To create `edge collection` use same endpoint `/_fabric/{fabric_name}/_api/colle
-        "direction": "out"
-    }
+# Graph Traversal
+# Note :- To use Outbound Traversal use direction: out and direction: in for Inbound Traversal
```
```suggestion
# Set `direction` to `out` or `in` for outbound or inbound traversal.
```

jamesm-macrometa commented on docs/collections/graphs/using-rest-api.md:
```diff
@@ -60,193 +60,196 @@ To create `edge collection` use same endpoint `/_fabric/{fabric_name}/_api/colle
-    # set dropCollection to False
-    params = {"dropCollection": True}
+# Delete Graph and Collections
+# Note:- If you want to delete just the graph and keep collections then
```
```suggestion
# Set `dropCollection` to `False` if you want to delete the graph but keep the collections.
```

jamesm-macrometa reviewed (changes requested):
A few minor changes, otherwise looks good!

jamesm-macrometa committed:
Merge branch 'master' into patch-5

jesusgonzalez12 committed:
Apply suggestions from code review

Co-authored-by: James <86624594+jamesm-macrometa@users.noreply.github.com>

jesusgonzalez12 commented:
Thanks for the review.
Updated with suggestions.

koshy-thomas commented on docs/collections/graphs/using-rest-api.md:
```diff
@@ -60,193 +60,196 @@ To create `edge collection` use same endpoint `/_fabric/{fabric_name}/_api/colle
+# Create a HTTPS Session Part 1
+# The following is required when accessing Macrometa with your email and password instead of an API key or JWT token
+"""""
+url = f"{FED_URL}/_open/auth"
```
Oops I might have missed this while reviewing the audit code... but we don't need to call /_open/auth endpoint here since we are using Api-key now, auth endpoint is used to get JWT token by logging in from email and password. Really sorry for the late comment.

jesusgonzalez12 replied:
Hi Koshy.

Yes, I left that in so a user can use both possibilities, either an API key or email/pass. The latter is behind a multi-line comment, beginning with the URL variable... mmm constant. I'll check for PEP8 compliance on this part of the code.

koshy-thomas replied:
<img width="320" alt="image" src="https://user-images.githubusercontent.com/107413064/185224215-0d2cf82f-2828-472b-9f11-4e7f7fad8a90.png">
I guess you added 2 extra double quotes so I guess its not a comment anymore?

jesusgonzalez12 replied:
OK, that part of the code has been updated.
Got a couple of unneeded double quotes.

Thanks for your review Koshy.

koshy-thomas commented on docs/collections/graphs/using-rest-api.md:
```diff
@@ -60,193 +60,196 @@ To create `edge collection` use same endpoint `/_fabric/{fabric_name}/_api/colle
+API_KEY = "my IP key" # Change to your API key
+#EMAIL = "my email"
+#PASSWORD = "my password"
+#AUTH_TOKEN = "bearer " # Do not change
```
we should not keep this email,pass, authtoken commented since we are not showing the user any alternative way to use them here. if we are using APikey  then we should remove these other methods. Also API_KEY should have sample like API_KEY = "apikey yourapikey" since "apikey" keyword is needed and user doesn't know that. Really sorry for being so late.

jesusgonzalez12 replied:
Hi. I left these available in case the user chooses that method instead of the API key way. The user should also uncomment the multi-line comment beginning at line 84-85 "url = f"{FED_URL}/_open/auth"".

The "apikey" keyword is not needed as that works on my tests.

Thanks for your comments.

jesusgonzalez12 committed:
using-rest-api.md updated

Commented code updated to comply with PEP8 standard.
Removed unneeded extra double quotes that may be preventing setting a multi-line comment.
This commented code allows the user to change the authentication method from API key to an email/pass method

koshy-thomas committed:
Merge branch 'master' into patch-5

koshy-thomas committed:
Merge branch 'master' into patch-5
